server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -21,23 +21,32 @@
   protected void doGet(HttpServletRequest req, HttpServletResponse res)
       throws ServletException, IOException {
 
-    String videoName = req.getParameter("name");
+    String input = req.getParameter("name");
 
-    assert videoName == null : "Something went wrong with sending to backend.";
+    assert input == null : "Something went wrong with sending to backend.";
 
     // Use like this: {url_parameter, value}
     HashMap<String, String> videoArgs = new HashMap<>();
     HashMap<String, String> commentArgs = new HashMap<>();
     HashMap<String, String> nameArgs = new HashMap<>();
 
-    videoArgs.put("q", videoName);
-    videoArgs.put("type", "video");
-    String videoIdJson = new YoutubeRequest("search", videoArgs).getResult();
-    String videoId = getVideoId(videoIdJson);
+    String videoId = input; // assume user enters id
 
     commentArgs.put("part", "snippet");
-    commentArgs.put("videoId", videoId);
-    String commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+    commentArgs.put("videoId", input);
```

Comment:
Are video id's a standard length/format? You can likely skip most "search strings" just by checking if the video id is 11 characters with no spaces.

Reply:
I'm not sure I'm following...what do you mean by skip most search strings?
I also though of checking the id w/no spaces, but then some channels like "deestroying", "RomanAtwood", or any 11 word video would match.

Reply:
Yeah, but looking up a video with id="RomanAtwood" should return a 404, and then you can fall back on the default searching behavior. As it stands, if I send you a video name, you're going to make 2 requests to youtube anyways (one as if I sent an id and another as if I sent the video name), so this would be be a net positive in saving you quota.

Reply:
Ah makes sense

Reply:
Can you look at what I did to see if it's what you were getting at?